crates/chain/src/tx_graph.rs

```diff
@@ -187,38 +373,118 @@ impl TxGraph {
         }
         new
     }
+
+    /// Returns the resultant [`Additions`] if the given `txout` is inserted at `outpoint`. Does not
+    /// mutate `self`.
+    ///
+    /// Inserting floating txouts are useful for determining fee/feerate of transactions we care
+    /// about.
+    ///
+    /// The [`Additions`] result will be empty if the `outpoint` (or a full transaction containing
+    /// the `outpoint`) already existed in `self`.
+    pub fn insert_txout_preview(&self, outpoint: OutPoint, txout: TxOut) -> Additions<A> {
+        let mut update = Self::default();
+        update.txs.insert(
+            outpoint.txid,
+            (
+                TxNodeInternal::Partial([(outpoint.vout, txout)].into()),
+                BTreeSet::new(),
+                0,
+            ),
+        );
+        self.determine_additions(&update)
+    }
+
     /// Inserts the given [`TxOut`] at [`OutPoint`].
     ///
-    /// Note this will ignore the action if we already have the full transaction that the txout is
-    /// alleged to be on (even if it doesn't match it!).
-    pub fn insert_txout(&mut self, outpoint: OutPoint, txout: TxOut) -> Additions {
+    /// This is equivalent to calling [`insert_txout_preview`] and [`apply_additions`] in sequence.
+    ///
+    /// [`insert_txout_preview`]: Self::insert_txout_preview
+    /// [`apply_additions`]: Self::apply_additions
+    pub fn insert_txout(&mut self, outpoint: OutPoint, txout: TxOut) -> Additions<A> {
         let additions = self.insert_txout_preview(outpoint, txout);
         self.apply_additions(additions.clone());
         additions
     }
 
+    /// Returns the resultant [`Additions`] if the given transaction is inserted. Does not actually
+    /// mutate [`Self`].
+    ///
+    /// The [`Additions`] result will be empty if `tx` already exists in `self`.
+    pub fn insert_tx_preview(&self, tx: Transaction) -> Additions<A> {
+        let mut update = Self::default();
+        update
+            .txs
+            .insert(tx.txid(), (TxNodeInternal::Whole(tx), BTreeSet::new(), 0));
+        self.determine_additions(&update)
+    }
+
     /// Inserts the given transaction into [`TxGraph`].
     ///
     /// The [`Additions`] returned will be empty if `tx` already exists.
-    pub fn insert_tx(&mut self, tx: Transaction) -> Additions {
+    pub fn insert_tx(&mut self, tx: Transaction) -> Additions<A> {
         let additions = self.insert_tx_preview(tx);
         self.apply_additions(additions.clone());
         additions
     }
 
+    /// Returns the resultant [`Additions`] if the `txid` is set in `anchor`.
+    pub fn insert_anchor_preview(&self, txid: Txid, anchor: A) -> Additions<A> {
```

Comment:
Do we need all these "preview" methods now. We added  them for consistency with chain graph/sparse chain. But now we have moved to monotone structs dominating everything I don't think we need the preview stuff (which is only useful in non-monotone situations where you *might* care about writing the changeset to disk before applying it).

Reply:
Let's remove them in the PR that removes the old structures?